cpp/src/binaryop/compiled/struct_binary_ops.cuh

```diff
@@ -93,9 +93,17 @@ void apply_struct_binary_op(mutable_column_view& out,
       out.end<bool>(),
       device_comparison_functor{optional_iter, is_lhs_scalar, is_rhs_scalar, device_comparator});
   };
-  is_any_v<BinaryOperator, ops::LessEqual, ops::GreaterEqual>
-    ? tabulate_device_operator(table_comparator.less_equivalent(comparator_nulls, comparator))
-    : tabulate_device_operator(table_comparator.less(comparator_nulls, comparator));
+  if (cudf::get_nested_columns(tlhs).size() > 0 || cudf::get_nested_columns(trhs).size() > 0) {
```

Comment:
I wish we had a better way to do this that didn't require allocating a new vector of columns only to check the size. Something like `std::any_of` with a predicate that can exit early would be preferable. I see this pattern of using `get_nested_columns` to check its size in a few places. (In fact, it seems that the vector of columns is _never_ used and it is _only_ used for checking non-emptiness! See Evidence below) Maybe we need to add a `table_view::has_nested_columns` for this purpose in a later PR.

<details><summary>Evidence</summary>

```cpp
$ rg get_nested_columns
cpp/tests/table/experimental_row_operator_tests.cu
61:  if (cudf::get_nested_columns(input).size() > 0) {
95:  if (cudf::get_nested_columns(lhs).size() > 0 || cudf::get_nested_columns(rhs).size() > 0) {

cpp/src/table/row_operators.cu
362:  if (cudf::get_nested_columns(t).size() > 0) {

cpp/src/table/table_view.cpp
117:std::vector<column_view> get_nested_columns(table_view const& table)

cpp/include/cudf/table/table_view.hpp
345:std::vector<column_view> get_nested_columns(table_view const& table);

cpp/src/search/search_ordered.cu
76:  if (cudf::get_nested_columns(haystack).size() > 0 ||
77:      cudf::get_nested_columns(needles).size() > 0) {

cpp/src/binaryop/compiled/struct_binary_ops.cuh
96:  if (cudf::get_nested_columns(tlhs).size() > 0 || cudf::get_nested_columns(trhs).size() > 0) {

cpp/src/sort/sort_impl.cuh
130:  if (cudf::get_nested_columns(input).size() > 0) {
```
</details>

Reply:
I didn't realize until later in my review that this function `table_view::get_nested_columns` is being newly introduced. Could we instead write a _free_ function `cudf::has_nested_columns(table_view)`?

Reply:
Yes, this method was only introduced in this PR. I chose to write it this way because the most comparable API is `has_nullable_columns`, and because the current approach seemed more generalizable than a `has_nested_columns`. However, since nothing else currently makes use of either, I would instead propose a compromise: let's move this to `cudf::detail::has_nested_columns`. Then, depending on what future use case emerge (if any) we can expose the appropriate API publicly.

Reply:
Hmm it's a little less obvious of a choice here than I thought. I was hoping to avoid prematurely an extra API, but since the comparators are public calling code (like in `experimental_row_operator_tests.cu`) needs to be able to use this. Since we are looking to switch to using an alternate dispatch method that doesn't require the call site to have this information, though, I propose that we still make this a detail function and use it in our test for now. We can reevaluate based on how the refactor to a visitor API goes. Thoughts?

Reply:
I've implemented the detail API for now as I proposed. Let me know if you'd like me to change it.

Reply:
That approach sounds good to me.